### PR TITLE
[contract_manager] support for executor

### PR DESCRIPTION
**Pull request by 0xfirefist**

### CODE REVIEW

File: contract_manager/src/contracts/evm.ts [hunk 1]
```diff
@@ -413,6 +473,16 @@ export class EvmEntropyContract extends Storable {
     return new web3.eth.Contract(EXTENDED_ENTROPY_ABI, this.address);
   }
 
+  async getExecutorContract() {
```
Comment (m30m) [inline contract_manager/src/contracts/evm.ts:476]: Can you move executor functions to a separate `EvmExecutor` class?

File: contract_manager/src/executor.ts [hunk 2]
```diff
@@ -12,7 +13,14 @@ export async function executeVaa(senderPrivateKey: PrivateKey, vaa: Buffer) {
   const parsedVaa = parseVaa(vaa);
   const action = decodeGovernancePayload(parsedVaa.payload);
   if (!action) return; //TODO: handle other actions
-  for (const contract of Object.values(DefaultStore.contracts)) {
+
+  let contracts:
+    | Record<string, PriceFeedContract>
+    | Record<string, EvmEntropyContract>;
+  if (action instanceof EvmExecute) contracts = DefaultStore.entropy_contracts;
```
Comment (m30m) [inline contract_manager/src/executor.ts:20]: for EvmExecute action we don't need to loop over the contracts. We find the EvmChain via targetChainId and create a EvmExecutor class with the specified chain and `action.executorAddress` and then run the vaa on it.

**Review by m30m — APPROVED** [review]
Thanks, please make the changes before merging.

File: contract_manager/src/executor.ts [hunk 3]
```diff
@@ -1,7 +1,53 @@
...
+} from "xc_admin_common";
 import { DefaultStore } from "./store";
-import { PrivateKey } from "./base";
+import { PrivateKey, TxResult } from "./base";
+import { EvmExecutorContract } from "./contracts";
+import { EvmChain } from "./chains";
+
+// TODO: A better place for this would be `base.ts`. That will require
+// significant refactor. Todo in separate PR.
+interface GovernanceContract {
```
Comment (m30m) [inline contract_manager/src/executor.ts:14]: why not just use PriceFeedContract?

File: contract_manager/src/executor.ts [hunk 4]
```diff
@@ -1,7 +1,53 @@
...
+  getId(): string;
+  getGovernanceDataSource(): Promise<DataSource>;
+  getLastExecutedGovernanceSequence(): Promise<number>;
+  executeGovernanceInstruction(
+    senderPrivateKey: PrivateKey,
+    vaa: Buffer
+  ): Promise<TxResult>;
+}
+
+async function execute(
```
Comment (m30m) [inline contract_manager/src/executor.ts:24]: rename to something more specific like `executeForPriceFeedContract`?

**Review by m30m — APPROVED** [review]
LGTM, made minor comments

File: contract_manager/src/executor.ts (hunk 3 above)
Comment (0xfirefist) [reply contract_manager/src/executor.ts:14]: The EvmExecutorContract doesn't and shouldn't implement PriceFeedContract.

**Pull request merged**